src/directives/modal.js

```diff
@@ -31,91 +31,159 @@ goog.require('ngeo');
-      // other elements even if in a positioned element initially.
-      angular.element(document.body).append(modal);
-
-      modal.find('.modal-dialog').draggable();
-
-      ngModelController.$render = function() {
-        modal.modal(ngModelController.$viewValue ? 'show' : 'hide');
-      };
-
-      modal.on('shown.bs.modal hidden.bs.modal', (e) => {
-        const type = e.type;
-        goog.asserts.assert(type == 'shown' || type == 'hidden');
-        scope.$apply(() => {
-          ngModelController.$setViewValue(type == 'shown');
-        });
+    this.destroyContent;
+
+    /**
+     * @private
+     * @type {boolean}
+     */
+    this.resizable;
+
+    /**
+     * @type {angular.NgModelController|null}
+     * @export
+     */
+    this.ngModel;
+  }
+
+  /**
+   * @export
+   */
+  $onInit() {
+    this.modal = this.$element.children();
+
+    this.destroyContent = !!this.destroyContent;
+    this.resizable = !!this.resizable;
+
+    this.childScope = this.$scope.$new();
+
+    // move the modal to document body to ensure that it is on top of
+    // other elements even if in a positioned element initially.
+    angular.element(document.body).append(this.modal);
+
+    this.modal.find('.modal-dialog').draggable();
+
+    this.ngModel.$render = function() {
+      this.modal.modal(this.ngModel.$viewValue ? 'show' : 'hide');
+    };
+
+    this.modal.on('shown.bs.modal hidden.bs.modal', (e) => {
+      const type = e.type;
+      goog.asserts.assert(type == 'shown' || type == 'hidden');
+      this.$scope.$apply(() => {
+        this.ngModel.$setViewValue(type == 'shown');
       });
+    });
 
-      if (destroyContent) {
-        modal.on('hide.bs.modal', onHide);
-        modal.on('show.bs.modal', onShow);
+    if (this.destroyContent) {
+      this.modal.on('hide.bs.modal', this.onHide.bind(this));
+      this.modal.on('show.bs.modal', this.onShow.bind(this));
+    } else {
+      if (this.resizable) {
+        this.modal.find('.modal-content').resizable().append(this.$transclude());
       } else {
-        if (resizable) {
-          modal.find('.modal-content').resizable().append(transcludeFn());
-        } else {
-          modal.find('.modal-content').append(transcludeFn());
-        }
+        this.modal.find('.modal-content').append(this.$transclude());
       }
+    }
+  }
 
-      function onShow(e) {
-        childScope = scope.$new();
-        transcludeFn(childScope, (clone) => {
-          if (resizable) {
-            modal.find('.modal-content').resizable().append(clone);
-          } else {
-            modal.find('.modal-content').append(clone);
-          }
-        });
+  /**
+   * @private
+   */
+  onShow() {
+    this.childScope = this.$scope.$new();
+    // * @param {!function(!angular.Scope=, !function(Element)=)=} transcludeFn
+    // function((angular.Scope|null)=, function(angular.Scope, Element): ?=, (Element|null)=, string=): ?
```

Comment:
`// function((angular.Scope|null)=, function(angular.Scope, Element): ?=, (Element|null)=, string=): ?`

I'm a bit confused by this annotation.

Reply:
It was an old annotation, I just forgot to remove it. Sorry for that)